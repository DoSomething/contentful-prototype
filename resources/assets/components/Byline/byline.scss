@import '../../scss/next-toolbox';

.byline {
  .figure {
    width: auto;
    display: inline-block;
    margin-bottom: 0;
  }

  .share {
    vertical-align: top;
    float: right;
  }
}
