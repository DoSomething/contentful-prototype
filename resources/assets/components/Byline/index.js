/**
 * The <Byline /> component accepts details
 * about an author and the option to add a
 * Facebook share.
 */
export default from './Byline';
