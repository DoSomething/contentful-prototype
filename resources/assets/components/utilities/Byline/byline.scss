@import '../../../scss/next-toolbox';

.byline {
  .figure {
    width: auto;
    display: inline-block;
  }

  .share {
    vertical-align: top;
    float: right;
  }
}
