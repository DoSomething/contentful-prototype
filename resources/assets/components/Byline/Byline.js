import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { Figure } from '../Figure';
import DEFAULT_AVATAR from './default-avatar.png';

import './byline.scss';

const Byline = ({ author, jobTitle, avatar, share, className }) => (
  <div className={classnames('byline', className)}>
    <Figure
      size="small"
      alignment="left"
      image={avatar}
      alt={`picture of ${author}`}
      imageClassName="avatar"
      className="margin-bottom-none"
    >
      <strong>{author}</strong>
      <br />
      <p className="footnote">{jobTitle}</p>
    </Figure>
    {share}
  </div>
);

Byline.propTypes = {
  author: PropTypes.string,
  className: PropTypes.string,
  jobTitle: PropTypes.string,
  avatar: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
  share: PropTypes.node,
};

Byline.defaultProps = {
  author: 'Puppet Sloth',
  className: null,
  jobTitle: 'DoSomething.org Staff',
  avatar: DEFAULT_AVATAR,
  share: null,
};

export default Byline;
