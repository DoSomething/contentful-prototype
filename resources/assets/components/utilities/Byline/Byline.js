import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import { Figure } from '../../Figure';
import DEFAULT_AVATAR from './default-avatar.png';

import './byline.scss';

/**
 * The <Byline /> component accepts details
 * about an author and the option to add a
 * Facebook share.
 */
const Byline = ({ author, jobTitle, photo, share, className }) => (
  <div className={classnames('byline', className)}>
    <Figure
      size="small"
      alignment="left"
      image={photo}
      alt={`picture of ${author}`}
      imageClassName="avatar"
      className="margin-bottom-none"
    >
      <strong>{author}</strong>
      <br />
      <p className="footnote">{jobTitle}</p>
    </Figure>
    {share}
  </div>
);

Byline.propTypes = {
  author: PropTypes.string,
  className: PropTypes.string,
  jobTitle: PropTypes.string,
  photo: PropTypes.string,
  share: PropTypes.node,
};

Byline.defaultProps = {
  author: 'Puppet Sloth',
  className: null,
  jobTitle: 'DoSomething.org Staff',
  photo: DEFAULT_AVATAR,
  share: null,
};

export default Byline;
