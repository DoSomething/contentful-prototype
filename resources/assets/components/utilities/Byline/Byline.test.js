import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Byline from './Byline';

test('Byline default snapshot test', () => {
  const component = shallow(<Byline />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});

test('Byline with props snapshot test', () => {
  const component = shallow(
    <Byline
      author="BraÃ¼mhilda Snosages"
      jobTitle="Campaign Tester"
      avatar="http://placeimg.com/150/150/people"
    />,
  );
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
