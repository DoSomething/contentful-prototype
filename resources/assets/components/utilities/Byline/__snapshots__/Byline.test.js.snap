// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Byline default snapshot test 1`] = `
<div
  className="byline"
>
  <Figure
    alignment="left"
    alt="picture of Puppet Sloth"
    className="mb-0"
    image="http://placekitten.com/300/300"
    imageClassName="avatar"
    size="small"
  >
    <strong>
      Puppet Sloth
    </strong>
    <br />
    <p
      className="text-sm text-gray-600"
    >
      DoSomething.org Staff
    </p>
  </Figure>
</div>
`;

exports[`Byline with props snapshot test 1`] = `
<div
  className="byline"
>
  <Figure
    alignment="left"
    alt="picture of Braümhilda Snosages"
    className="mb-0"
    image="http://placeimg.com/150/150/people"
    imageClassName="avatar"
    size="small"
  >
    <strong>
      Braümhilda Snosages
    </strong>
    <br />
    <p
      className="text-sm text-gray-600"
    >
      Campaign Tester
    </p>
  </Figure>
</div>
`;
